packages/react-components/src/components/ChatMessage/ChatMessageContent.tsx

```diff
@@ -84,12 +83,74 @@ const MessageContentAsRichTextHTML = (props: ChatMessageContentProps): JSX.Eleme
     });
   }, [props]);
 
+  const processNodeDefinitions = ProcessNodeDefinitions();
+  const processingInstructions: ProcessingInstructionType[] = [
+    // Mention render override if needed
+    {
+      shouldProcessNode: (node) => {
+        /* @conditional-compile-remove(mention) */
+        if (props.mentionDisplayOptions?.onRenderMention) {
+          // Override the handling of the <msft-mention> tag in the HTML if there's a custom renderer
+          return node.name === 'msft-mention';
+        }
+        return false;
+      },
+      processNode: (node) => {
+        /* @conditional-compile-remove(mention) */
+        if (props.mentionDisplayOptions?.onRenderMention) {
```

Comment:
We can pass this part as a paramenter
I would imagine we just need to call a single util function here like 
parseHtmlNode([processors]);

Then we could have 2 processors for now
const mentionProcessors = createMentionProcessor(props.mentionDisplayOptions.onRenderMention);
const inlineImageProcessors = createInlineImageProcessor();

And wrap these 2 create functions with useMemo to avoid duplication and unnecessary re-render

Reply:
inlineImageProcessors could be a const value outside of the function, so no need for useMemo